obsercer_pattern.java

```diff
@@ -0,0 +1,75 @@
+import java.util.ArrayList;
+import java.util.List;
+
+// Subject interface
+interface Subject {
+    void addObserver(Observer observer);
+    void removeObserver(Observer observer);
+    void notifyObservers();
+}
+
+// Concrete Subject
+class ConcreteSubject implements Subject {
+    private List<Observer> observers = new ArrayList<>();
+    private int state;
+
+    public int getState() {
+        return state;
+    }
+
+    public void setState(int state) {
+        this.state = state;
+        notifyObservers();
```

Comment:
Although automatically calling the `notifyObservers` method is completely acceptable, depending on use cases it may be better to manually call it,

Reply:
@Dhuds1 Thank you for your suggestion let me call it manually.